python/ray/data/tests/test_dataset_pipeline.py

```diff
@@ -183,11 +186,23 @@ def test_basic_pipeline(ray_start_regular_shared):
     pipe = ds.window(blocks_per_window=1)
     assert str(pipe) == "DatasetPipeline(num_windows=10, num_stages=2)"
     assert pipe.count() == 10
+    pipe = ds.window(blocks_per_window=1)
+    pipe.show()
```

Comment:
Note that this will print to stdout, is that intended? If you're looking to trigger execution, doing something like
```suggestion
    assert pipe.take() == list(range(10))
```
would be better.